pkg/kubelet/dockertools/manager.go

```diff
@@ -1919,7 +1919,7 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 			continue
 		}
 
-		if container.SecurityContext != nil && container.SecurityContext.RunAsNonRoot {
+		if container.SecurityContext != nil && container.SecurityContext.RunAsNonRoot != nil && *container.SecurityContext.RunAsNonRoot {
```

Comment:
We are getting this type of code sprinkled in the manager quite a bit.  I think we should start making utilities in the provider utils for things like `isRunAsNonRoot(sc)`